ProtectionScan/Features/MainFeature.cs

```diff
@@ -248,9 +258,62 @@ private void WriteProtectionResultJson(string path, Dictionary<string, List<stri
                 // Attempt to open a protection file for writing
                 using var jsw = new StreamWriter(File.OpenWrite($"protection-{DateTime.Now:yyyy-MM-dd_HHmmss.ffff}.json"));
 
-                // Create the output data
                 var jsonSerializerOptions = new System.Text.Json.JsonSerializerOptions { WriteIndented = true };
-                string serializedData = System.Text.Json.JsonSerializer.Serialize(protections, jsonSerializerOptions);
+                string serializedData;
+                if (Nested)
+                {
+                    // A nested dictionary is used to achieve proper serialization.
+                    var nestedDictionary = new Dictionary<string, object>();
+                    var trimmedPath = path.TrimEnd(['\\', '/']); 
+                    
+                    // Sort the keys for consistent output
+                    string[] keys = [.. protections.Keys];
+                    Array.Sort(keys);
+                    
+                    var modifyNodeList = new List<(Dictionary<string, object>, string, string[])>();
+
+                    // Loop over all keys
+                    foreach (string key in keys)
+                    {
+                        // Skip over files with no protection
+                        var value = protections[key];
+                        if (value.Count == 0)
+                            continue;
+
+                        // Sort the detected protections for consistent output
+                        string[] fileProtections = [.. value];
+                        Array.Sort(fileProtections);
+
+                        // Inserts key and protections into nested dictionary, with the key trimmed of the base path.
+                        InsertNode(nestedDictionary, key.Substring(trimmedPath.Length), fileProtections, modifyNodeList);
+                    }
+
+                    // Adds the non-leaf-node protections back in
+                    for (int i = 0; i < modifyNodeList.Count; i++)
+                    {
+                        string part = modifyNodeList[i].Item2;
+                        string[] nodeProtections = modifyNodeList[i].Item3;
+                        var copyDictionary = new Dictionary<string, object>((Dictionary<string, object>)modifyNodeList[i].Item1[part]);
```

Comment:
Is there any chance that `Item1[part]` could not exist? Doing a `TryGet` might be necessary if there is some doubt, just so this doesn't throw an exception if there's an issue.